jetbrains-core/src/software/aws/toolkits/jetbrains/utils/ui/UiUtils.kt

```diff
@@ -36,15 +36,20 @@ fun <T> ComboBox<T>.populateValues(
     block: () -> Collection<T>
 ) {
     ApplicationManager.getApplication().executeOnPooledThread {
+        val previouslySelected = this.model.selectedItem
+        val previousState = this.isEnabled
+        this.model.selectedItem = "Loading..."
+        this.isEnabled = false
         val values = block()
         ApplicationManager.getApplication().invokeLater({
             val model = this.model as DefaultComboBoxModel<T>
-            val previouslySelected = model.selectedItem
             model.removeAllElements()
             values.forEach { model.addElement(it) }
             this.selectedItem = if (forceSelectDefault || previouslySelected == null) default else previouslySelected
             if (updateStatus) {
                 this.isEnabled = values.isNotEmpty()
+            } else {
+                this.isEnabled = previousState
```

Comment:
Should this be in a finally block in case `block()` fails in a bad way?

Reply:
This could be a bug for the error handling when loading resources. We need to create a separate issue for handling this. The correct behavior in this case should be showing a special item `Failed` and a validation failure should be returned. Created an issue #535 for tracking this.